Задача 2. Исследовалось влияние препарата на уровень давления пациентов. Сначала
измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть
ли статистически значимые различия между измерениями давления? В выборках не соблюдается условие нормальности.
1е измерение до приема препарата: 150, 160, 165, 145, 155
2е измерение через 10 минут: 140, 155, 150, 130, 135
3е измерение через 30 минут: 130, 130, 120, 130, 125

In [2]:
import numpy as np
from scipy import stats

In [3]:
sample1 = np.array([150, 160, 165, 145, 155])
sample2 = np.array([140, 155, 150, 130, 135])
sample3 = np.array([130, 130, 120, 130, 125])

Принимаем α=0,05.

In [4]:
stats.friedmanchisquare(sample1, sample2, sample3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Т.к. pvalue=0.00831683351100441 < α=0,05, можно говорить о том что статистически значимые различия между измерениями давления есть.